fuzz_test.go

```diff
@@ -0,0 +1,109 @@
+package dns
+
+import (
+	"net"
+	"testing"
+)
+
+// TestPackDataOpt tests generated using fuzz.go and with a message pack
+// containing the following bytes:
+// "0000\x00\x00000000\x00\x00/00000" +
+// "0\x00\v\x00#\b00000000\x00\x00)000" +
+// "000\x00\x1c00\x00\x0000\x00\x01000\x00\x00\x00\b" +
+// "\x00\v\x00\x02\x0000000000"
+// That bytes sequence created the overflow error.
+func TestPackDataOpt(t *testing.T) {
```

Comment:
could you move this to #966 ?

Reply:
This is already in #966. This PR is made of 2 diffs, the one from dataopt (#966) and the nsec one. Because fuzz_test.go was newly created, it would conflict. It will be easier to just rebase the last commotion top of 966 once it is landed (which is what I did in this diff). Essentially what only matters in this PR is the top diff.